### DISCUSSION TITLE
conversation_id under get_all_tweets not returning entire conversation thread

### DISCUSSION THREAD

**Discussion by alexikim** [cjbarrie/academictwitteR#189] [Q&A]
I am using R to fetch all comments in conversation with a specific twitter status id. The status ID has 1492 quote_tweets, and only 856 commented tweets are returning. I am on an academic API so should have access to the full archive. Any thoughts?

This is the tweet (status ID: 789903797803970560) I am interested in https://twitter.com/RussianEmbassy/status/789903797803970560. I am using the academictwitteR package in R; code below.

get_all_tweets(start_tweets = tweet_metrics$tweet_time,
end_tweets = current_time,
bearer_token = get_bearer(),
n = tweet_metrics$quote_count,
conversation_id = tweet_status_id)

The tweet_time, current_time, quote_count, and tweet_status_id change accordingly depending on the the tweet of interest. For this one, it is as follows: “2016-10-22T18:58:35Z”, “2021-07-09T22:36:59Z”, “1492”, and “789903797803970560”, respectively.

One thing I notice is that when I make n = 1492 (the quote_count), I actually only get about 500 comments. When I enter a much larger number, say 50000 as the n max, then I get 856. Not sure what to think of this... 

How do I get the whole record? Thanks!

**Comment by chainsawriot** [comment]
@alexikim Thanks for your question.
You should have gotten an answer [from Igor, right](https://twittercommunity.com/t/conversation-id-on-r-academictwitter-package-not-generating-all-threads/156674)? What a nice guy! 

Actually, I can get 856 with `n = 1492`. Okay, it could be the problem with the software. But I think it is likely not. It's probably neither a problem with Twitter API too. As Igor has said, "1492" is actually not the number of tweets one is expected to retrieve. It is because the number includes private tweets, deleted tweets, tweets from suspended accounts etc. So, with your highly controversial tweets, it is not a surprise to have a lot of missing tweets. Twitter data do decay. Be prepared to [cite this paper](https://doi.org/10.3389/fdata.2019.00013).

**Comment by alexikim** [comment]
@chainsawriot 
Thanks for this response! Along with what Igor said, I cross checked the
text output of conversation_id, and I see that that conversation_id will
only pull *replies* to a tweet, but it does not fetch *quote tweets*.

To get quote tweets, I would need something that is along the lines of
(url:789903797803970560 is:quote) as a specification option in get_all_tweets(). Is there
any functionality that would do this in the academictwitteR? When I do is_quote = TRUE, it is only pulling 10 tweets for this specific tweet, even though there were at one point 1492 quote tweets (I have n=1500). Is this just an API limit?

Thanks for all this help! I really appreciate it.

**Reply by chainsawriot** [reply]
@alexikim Unfortunately, it can't. But here is a quick fix. Igor's approach is great, but it might has a drawback (e.g. a quoted tweets with a URL such as "google.com/789903797803970560" will also be returned). But I think you can cross-check this and see if it is really the case.

``` r
require(academictwitteR)
#> Loading required package: academictwitteR
build_query2 <- function(conversation_id) {
    glue::glue("conversation_id:{conversation_id} OR (url:{conversation_id} is:quote)", conversation_id = conversation_id)
}
x <- get_all_tweets(build_query2(conversation_id = "789903797803970560"), start_tweets = "2016-10-22T18:58:35Z", end_tweets = "2021-07-09T22:36:59Z", bearer_token = get_bearer(), n = Inf)
#> Warning: Recommended to specify a data path in order to mitigate data loss when
#> ingesting large amounts of data.
#> Warning: Tweets will not be stored as JSONs or as a .rds file and will only be
#> available in local memory if assigned to an object.
#> query:  conversation_id:789903797803970560 OR (url:789903797803970560 is:quote) 
#> Total pages queried: 1 (tweets captured this page: 495).
#> Total pages queried: 2 (tweets captured this page: 481).
#> Total pages queried: 3 (tweets captured this page: 496).
#> Total pages queried: 4 (tweets captured this page: 494).
#> Total pages queried: 5 (tweets captured this page: 489).
#> Total pages queried: 6 (tweets captured this page: 486).
#> Total pages queried: 7 (tweets captured this page: 489).
#> Total pages queried: 8 (tweets captured this page: 487).
#> Total pages queried: 9 (tweets captured this page: 480).
#> This is the last page for conversation_id:789903797803970560 OR (url:789903797803970560 is:quote) : finishing collection.
nrow(x)
#> [1] 4397
```

<sup>Created on 2021-07-11 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>
